runtime/foucoco/src/lib.rs

```diff
@@ -1320,7 +1332,27 @@ where
 		true
 	}
 }
-
+#[derive(Debug, Clone, PartialEq, Eq, codec::Encode, codec::Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub struct CoinInfo {
+	pub symbol: Vec<u8>,
+	pub name: Vec<u8>,
+	pub blockchain: Vec<u8>,
+	pub supply: u128,
+	pub last_update_timestamp: u64,
+	pub price: u128,
+}
+fn trim_trailing_zeros(slice: &[u8]) -> &[u8] {
+	let mut trim_amount = 0;
+	for el in slice.iter().rev() {
+		if *el == 0 {
+			trim_amount += 1;
+		} else {
+			break
+		}
+	}
+	&slice[..slice.len() - trim_amount]
+}
```

Comment:
Nit: Maybe add more new lines between the declarations.

Reply:
done. Also, moved them to runtime/common/chain_ext.rs
The contact (in the other PR) is now using the same types as the chain from that file instead of maintaining a duplicate.